WordPress/src/test/java/org/wordpress/android/ui/jetpack/scan/usecases/StartScanUseCaseTest.kt

```diff
@@ -0,0 +1,93 @@
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.whenever
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.InternalCoroutinesApi
+import kotlinx.coroutines.flow.toList
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Before
+import org.junit.Test
+import org.mockito.Mock
+import org.wordpress.android.BaseUnitTest
+import org.wordpress.android.fluxc.action.ScanAction.START_SCAN
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.fluxc.model.scan.ScanStateModel
+import org.wordpress.android.fluxc.store.ScanStore
+import org.wordpress.android.fluxc.store.ScanStore.OnScanStarted
+import org.wordpress.android.fluxc.store.ScanStore.ScanStartError
+import org.wordpress.android.fluxc.store.ScanStore.ScanStartErrorType
+import org.wordpress.android.test
+import org.wordpress.android.ui.jetpack.scan.usecases.StartScanUseCase.StartScanState
+import org.wordpress.android.util.NetworkUtilsWrapper
+
+@InternalCoroutinesApi
+class StartScanUseCaseTest : BaseUnitTest() {
+    private lateinit var useCase: StartScanUseCase
+    @Mock private lateinit var site: SiteModel
+    @Mock private lateinit var scanStateModel: ScanStateModel
+    @Mock lateinit var networkUtilsWrapper: NetworkUtilsWrapper
+    @Mock lateinit var scanStore: ScanStore
+
+    @Before
+    fun setup() {
+        useCase = StartScanUseCase(networkUtilsWrapper, scanStore)
+        whenever(networkUtilsWrapper.isNetworkAvailable()).thenReturn(true)
+        whenever(scanStore.getScanStateForSite(site)).thenReturn(scanStateModel)
+    }
+
+    @Test
+    fun `given site and no network, when scan is started, then NetworkUnavailable is returned`() = test {
+        whenever(networkUtilsWrapper.isNetworkAvailable()).thenReturn(false)
+
+        val result = useCase.startScan(site).toList(mutableListOf())
+
+        assertThat(result).contains(StartScanState.Failure.NetworkUnavailable)
+    }
+
+    @Test
+    fun `given site, when scan is started, then Success is returned on success`() = testWithSuccessResponse {
+        val result = useCase.startScan(site).toList(mutableListOf())
+
+        assertThat(result).contains(StartScanState.Success)
+    }
+
+    @Test
+    fun `given site, when scan is started, then scanning scan state is updated in db and returned`() =
+        testWithSuccessResponse {
+            val scanStateModel = ScanStateModel(state = ScanStateModel.State.IDLE)
+            whenever(scanStore.getScanStateForSite(any())).thenReturn(scanStateModel)
+
+            val result = useCase.startScan(site).toList(mutableListOf())
+
+            verify(scanStore).storeScanState(any())
+            assertThat(result).contains(
+                StartScanState.ScanningStateUpdatedInDb(scanStateModel.copy(state = ScanStateModel.State.SCANNING))
+            )
+        }
+
+    @Test
+    fun `given site, when scan is started, then RemoteRequestFailure is returned on failure`() =
```

Comment:
🔍 ❓ I'm not used to "given, when, then" naming. However, I'm wondering if something like this "given server is unavailable, when scan is started, then RemoteRequestFailure is returned" would be a bit clearer. Wdyt?

Reply:
Yes this looks more clearer. In 144e00a.